### ISSUE TITLE
About the network model and codes

### ISSUE THREAD

**Issue by song201216** [htzheng/CM-GAN-Inpainting#1]
Hello, your work is very amazing, but when will you release the network structure and training code?
Thanks！

**Comment by htzheng** [comment]
Hi @song201216 , thank you for your interest in this project. We plan to release the online demo soon for inference. Since the model was developed during my internship, I am still at the stage to request permission for releasing the code.

**Comment by alexmi256** [comment]
I hope you're successfull in your request to release. 
One thing I'm curious about is if you were able to test this with a faces dataset to see how it performs.
Also, how long does training take given the hardware that you had?

**Comment by htzheng** [comment]
> I hope you're successfull in your request to release. One thing I'm curious about is if you were able to test this with a faces dataset to see how it performs. Also, how long does training take given the hardware that you had?

Hi, thank you for your interest in our project! We will release the online demo of CM-GAN soon (probably in the next two months) and will most likely release the training code after releasing the online demo. Currently, we only test the model on Places2 but we are working on training an FFHQ model. Our model is trained on A100 GPUs for 1 week (similar to CoModGAN).

**Comment by styler00dollar** [comment]
How many GPUs were used?

**Comment by htzheng** [comment]
> How many GPUs were used?

It uses 8 gpus.

**Comment by roimulia2** [comment]
Any news about the pre-trained model? :) Loved the paper, would love to benchmark it.

**Comment by htzheng** [comment]
@roimulia2 Thank you for your interest in this work. I am gradually working towards releasing the model. Currently, the dataset, panoptic annotation, evaluation of CM-GAN, and the on-the-fly mask generation code are released. Will work on the online demo next.

**Comment by htzheng** [comment]
> I hope you're successfull in your request to release. One thing I'm curious about is if you were able to test this with a faces dataset to see how it performs. Also, how long does training take given the hardware that you had?

Thank you for raising this and it is a very good point. We are also interested in training a model on FFHQ and will possibly work on it later.

**Comment by surelon** [comment]
Dear author, hello! I am also interested in the network model of your article, may I ask whether it will be updated on Github after the license is released? I do object detection, would like to quote this article, if convenient, can you send a copy to sdaujdxy2016ks@163.com, thank you very much!

**Comment by raziehkaviani** [comment]
Thank you for the amazing work and impressive results. I am wondering that when will you release the code? Thanks much.

**Comment by shallweiwei** [comment]
@htzheng Hello, could you release the training and inference code ?

**Comment by htzheng** [comment]
Since my company is developing products with CM-GAN, they are still deciding when to release the code. Sorry about this...

**Comment by liangyuanyuan-lyy** [comment]
Hello, I am very interested in what you have published. Can I publish the training code of CM-GAN?

**Comment by Sanster** [comment]
@htzheng I am very interested in the Lama-OT model. Is it possible to release the weight file of this model? Thank you.

**Comment by lijain** [comment]
![image](https://github.com/htzheng/CM-GAN-Inpainting/assets/18046568/42becd0a-354e-4514-9d61-a21bf6a4eee1)

**Comment by lijain** [comment]
![image](https://github.com/htzheng/CM-GAN-Inpainting/assets/18046568/fc08e27e-5964-4313-9e5f-b77126213aad)
你们的工作非常有趣，不过对论文中的有些地方比较困惑，可以方便解答？

**Comment by lijain** [comment]
你们的工作非常有趣，不过对论文中的有些地方比较困惑，可以方便解答？

**Comment by lijain** [comment]
看了你们的文章，感觉很有趣，不过有几点不是很清楚，可以方便解答下？

**Comment by lijain** [comment]
看了你们的文章，感觉很有趣，不过有几点不是很清楚，可以方便解答下？
![image](https://github.com/htzheng/CM-GAN-Inpainting/assets/18046568/dc6b598a-5e95-4194-9f51-bdaafd48b9e0)
1 你们的Fout g也是会单独计算loss是？
2 第二decoder是不引入encoder的信息，只拿取第一个decoder的Fine?
